packages/angular/ssr/src/app.ts

```diff
@@ -113,3 +97,27 @@ export class AngularServerApp {
     return render(this, request, serverContext, requestContext);
   }
 }
+
+let angularServerApp: AngularServerApp | undefined;
+
+/**
+ * Creates an instance of `AngularServerApp` if it does not already exist.
+ * If an instance exists, it will return the existing one.
+ *
+ * @param options - (Optional) The configuration options for the Angular server app.
+ * @returns The created or existing instance of `AngularServerApp`.
+ */
+export function createAngularServerApp(options?: AngularServerAppOptions): AngularServerApp {
```

Comment:
Nit: `create` strongly implies that this function always creates an `AngularServerApp`, when it is actually reused most of the time. Maybe `getAngularServerApps` or `getOrCreateAngularServerApp` might be more clear?

Related: Is it confusing that `options` is ignored and might diverge from a previous invocation?

```typescript
const app1 = createAngularServerApp({ hooks: [someHook] });
const app2 = createAngularServerApp({ hooks: [someOtherHook] }); // `someOtherHook` is actually unused since we'll reuse `app1`.
```

Reply:
Maybe in the case the options differ we should create a new instance?